Bots/DotNet/Skills/CodeFirst/TeamsWaterfallSkillBot/Dialogs/ActivityRouterDialog.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.TraceExtensions;
+using Microsoft.Bot.Schema;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Attachments;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Auth;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Cards;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Delete;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Echo;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.FileUpload;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Proactive;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Sso;
+using Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs.Update;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+
+namespace Microsoft.BotFrameworkFunctionalTests.TeamsWaterfallSkillBot.Dialogs
+{
+    /// <summary>
+    /// A root dialog that can route activities sent to the skill to different sub-dialogs.
+    /// </summary>
+    public class ActivityRouterDialog : ComponentDialog
+    {
+        public ActivityRouterDialog(IConfiguration configuration, IHttpClientFactory clientFactory, IHttpContextAccessor httpContextAccessor)
+            : base(nameof(ActivityRouterDialog))
+        {
+            AddDialog(new CardDialog(httpContextAccessor, clientFactory));
+            AddDialog(new WaitForProactiveDialog(httpContextAccessor));
+            AddDialog(new AttachmentDialog(new Uri($"{httpContextAccessor.HttpContext.Request.Scheme}://{httpContextAccessor.HttpContext.Request.Host.Value}")));
+            AddDialog(new AuthDialog(configuration));
+            AddDialog(new SsoSkillDialog(configuration));
+            AddDialog(new EchoDialog());
+            AddDialog(new FileUploadDialog());
+            AddDialog(new DeleteDialog());
+            AddDialog(new UpdateDialog());
+
+            AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[] { ProcessActivityAsync }));
+
+            // The initial child Dialog to run.
+            InitialDialogId = nameof(WaterfallDialog);
+        }
+
+        private async Task<DialogTurnResult> ProcessActivityAsync(WaterfallStepContext stepContext, CancellationToken cancellationToken)
```

Comment:
There seems to be some overlap between the things the TeamsWaterfall can do and what the WaterfallSkill do, can we remove some of the redundant cases and keep only the things that are special for teams in this one?

Reply:
Yes and no. I'll work on building the demo so we can better address this issue.